### PR TITLE
win_service Added -Force option and minor cleanup

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
As per the discussion in the Windows user group this cleans up win_service so that it matches general standard and adds the optional parameter around whether to force a service to stop.

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME
win_service

##### ANSIBLE VERSION
```
ansible 2.3.0 (devel 771b14acb2) last updated 2017/03/14 17:33:12 (GMT +1000)
  config file = /mnt/c/dev/test/role/ansible.cfg
  configured module search path = [u'/mnc/c/dev/test/role/library']
  python version = 2.7.12 (default, Dec 14 2016, 21:05:07) [GCC 4.8.4]
```

### CODE REVIEW

**Comment by jborean93** [comment]
@nitzmahone @dagwieers @jhawkesworth

**Comment by dagwieers** [comment]
So you basically reimplemented #20539 and #22311 too :-)

I think it is unfortunate that effort is duplicated like this (because our time could be spend better) and I hope we can prevent this in the future.

shipit

**Review by dagwieers — APPROVED** [review]

**Comment by jborean93** [comment]
this change was from an action in today's meeting, I thought you wanted me to incorporate your changes into the one but I must have misinterpreted it.

**Comment by dagwieers** [comment]
@jborean93 Yeah, I have no problem with this. Fact is that if PRs would be merged quicker, you didn't have to do this as it would already be reviewed and merged using the normal process.

**Review by nitzmahone — CHANGES_REQUESTED** [review]
Hmm, wondering if there's a better name than `ignore_dependencies`, since we're not really "ignoring" them... We're stopping/restarting "dependent" services, not this service's dependencies- trying to figure out a concise way to name that... 

`allow_dependent_service_state_change` would be correct, but awfully long. Microsoft rarely trades brevity for clarity in Powershell, but in this case the terse `force` arg on `Stop-Service`/`Restart-Service` is not a name I want to emulate. `manage_dependent_services` maybe? Dunno, I'm open to suggestions, and if you're busy, I can do the rename if you like. Thoughts?

**Comment by jborean93** [comment]
Maybe `change_dependent_services`  as we are changing the state of the dependent services?

**Comment by nitzmahone** [comment]
Trying to find a word about twiddling their state...

**Comment by nitzmahone** [comment]
`X_dependent_services`, where X is:

manage
state_change
manipulate
modify

`state_change_dependent_services` is the most accurate of those, but funky English, which is why I kinda landed on `manage_dependent_services`, but I still don't *love* that either.

**Comment by nitzmahone** [comment]
I don't really want to split it into separate args for stop/restart, but if we *did*, `stop_dependent_services` and `restart_dependent_services` would be exactly correct... :(

**Comment by nitzmahone** [comment]
Which brings me back to `manage_dependent_services` as the best compromise.

**Comment by jborean93** [comment]
`allow_twiddling_of_dependent_services` :)

Could we put `force_dependent_services` as it matches the -Force side of the command, still not the best.

**Comment by nitzmahone** [comment]
Actually I don't hate `force_dependent_services`... That'll do- you want to make the change, or shall I?

**Comment by jborean93** [comment]
I'll make the change, there are some other typos I've found in the WMI error codes I want to fix up as well.

**Comment by nitzmahone** [comment]
OK, if you're doing typos, I saw a couple others too- go ahead and do the other stuff and I'll mark those up real quick.

File: lib/ansible/modules/windows/win_service.py [hunk 1]
```diff
@@ -60,18 +58,24 @@
...
-    required: False
     version_added: "2.3"
   display_name:
     description:
       - The display name to set for the service.
-    required: False
+    version_added: "2.3"
+  ignore_dependencies:
+    description:
+    - When stopping or restarting a service this will Force the service to stop
```
Comment (nitzmahone) [inline lib/ansible/modules/windows/win_service.py]: How about "If `true`, stopping or restarting a service with dependent services will force the dependent services to stop or restart also. If `false`, stopping or restarting a service with dependent services may fail.

**Review by nitzmahone — APPROVED** [review]

File: lib/ansible/modules/windows/win_service.py (hunk 1 above)
Comment (jborean93) [reply lib/ansible/modules/windows/win_service.py]: Have made the change, I wasn't fully aware of this behaviour but am now

**Comment by nitzmahone** [comment]
Noice, will wait for green from shippable and merge. Thanks!

**Pull request merged**